photutils/psf/photometry.py

```diff
@@ -306,16 +317,17 @@ def nstar(self, image, star_groups):
             Residual image.
         """
 
-        result_tab = Table([[], [], [], [], []],
-                           names=('id', 'group_id', 'x_fit', 'y_fit',
-                                  'flux_fit'),
-                           dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
-        star_groups = star_groups.group_by('group_id')
+        result_tab = Table()
+
+        for param_tab_name in self._pars_to_output.keys():
```

Comment:
I think this means you now to change the `nstar` docstring, right?  That is, it should also say that columns like `<param>_0` are required if you have any other parameters free?